pkg/apis/batch/v2alpha1/types.go

```diff
@@ -250,6 +250,16 @@ type CronJobSpec struct {
 	// JobTemplate is the object that describes the job that will be created when
 	// executing a CronJob.
 	JobTemplate JobTemplateSpec `json:"jobTemplate" protobuf:"bytes,5,opt,name=jobTemplate"`
+
+	// The number of successful finished jobs to retain.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// +optional
+	SuccessfulJobsHistoryLimit *int32 `json:"successfulJobsHistoryLimit,omitempty" protobuf:"varint,6,opt,name=successfulJobsHistoryLimit"`
```

Comment:
The vast majority of defaults are documented.  How else are users supposed to know what they are? 

Since that's potentially a data loss issue, you'll need to make very certain that the impacts of this are understood.  @kargakis did we do any prep for deployment cleanup?  At minimum the important release notes.